parachains/integration-tests/emulated/common/src/lib.rs

```diff
@@ -495,7 +455,510 @@ decl_test_sender_receiver_accounts_parameter_types! {
 	BridgeHubRococo { sender: ALICE, receiver: BOB },
 	BridgeHubWococo { sender: ALICE, receiver: BOB },
 	// Penpals
-	PenpalPolkadot { sender: ALICE, receiver: BOB },
-	PenpalKusama { sender: ALICE, receiver: BOB },
-	PenpalWestend { sender: ALICE, receiver: BOB }
+	PenpalPolkadotA { sender: ALICE, receiver: BOB },
+	PenpalPolkadotB { sender: ALICE, receiver: BOB },
+	PenpalKusamaA { sender: ALICE, receiver: BOB },
+	PenpalKusamaB { sender: ALICE, receiver: BOB },
+	PenpalWestendA { sender: ALICE, receiver: BOB },
+	PenpalRococoA { sender: ALICE, receiver: BOB }
+}
+
+pub mod events {
+	pub mod polkadot {
+		use crate::*;
+		type RuntimeEvent = <Polkadot as Chain>::RuntimeEvent;
+
+		// Dispatchable is completely executed and XCM sent
+		pub fn xcm_pallet_attempted_complete(expected_weight: Option<Weight>) {
```

Comment:
other thing is that I think those function looks the same,
and they could be write in generic way for `trait Chain`

Reply:
Now that we have chains impl, yes, I can move it to the impls.

About moving to `Chain`... `xcm-emulator` will not compile for someone that who is using a chain without one of the expected `RuntimeEvents`. I do not think it is a good idea or necessary. It would fail for those chains that use `xTokens` pallet instead of `XcmPallet` for example.

Reply:
I didn't want to, but I ended up using macros. It was too much repetition.

Reply:
yes, I got it that xcm-emulator is general, but there is also possiblity to create `CumulusChain / PolkadotLikeTelayChain` in `emulated/common` and so on